### ISSUE TITLE
Add support for ammunition and other consumables

### ISSUE THREAD

**Issue by xdy** [xdy/twodsix-foundryvtt#212]
**Is your feature request related to a problem? Please describe.**
From @marvin9257 on discord: "I ran my first session last night with the system under CEL and I thought it went well.  One feature the players asked for was a better way to track ammo usage.  Right now, there is only one field (ammo) that currently is the magazine size.  Either an generic counter or an additional field to track current ammo amount would be useful.  Even better if it could be modified right on the character sheet without having to open a separate edit window.  They are currently tracking this amount in notes.  Thanks."

**Describe the solution you'd like**
Add support for consumables, like dnd5e has.
Make items with consumable set, and that are loaded, consume it. (Default to consuming 1, add selection of number to advanced roll I guess?)

**Describe alternatives you've considered**
Keep taking notes!

**Additional context**
See dnd5e, search for ammo and consume to find the relevant bits of code.
Basically, add this to item template, and add support for it to rolls from items, I guess
        "consume": {
          "type": "",
          "target": null,
          "amount": null
        }

**Comment by marvin9257** [comment]
The SWN system in Foundry also has a nice implementation of a similar feature.  You could look there for ideas.  Thanks

**Comment by jonepatr** [comment]
I could have a look at this feature!

**Comment by xdy** [comment]
Go ahead!

**Comment by jonepatr** [comment]
Hm, adding the consumption of ammunition might bloat the dice roller. 

I'm considering some refactoring and moving things around. 
What about having a method called `performAttack` in the `TwodsixItem` class? This method could then also consider relevant weapons traits, and properties of the weapon (such as how many bullets to consume, if the weapon is bulky apply relevant DMs, etc.). Furthermore we could move the skill roll method to the actor class for the actual skill roll, and the `rollDamage` would be moved to the item class (and then again, we can use the traits, if the weapon has AP, we would write that in the chat).

**Comment by xdy** [comment]
TwodsixRolls is a monstrosity, yeah.
The main problem I have with your suggestion is that it's the result of me looking at rolls code all over the place (as in, in the Actor and Item classes), often duplicated.
But, the current situation isn't good either...
So, yeah, try it out, see how it feels.
